src/Text2Image/T2IParamInput.cs

```diff
@@ -463,6 +469,32 @@ static T2IParamInput()
         PromptTagLengthEstimators["embed"] = PromptTagLengthEstimators["preset"];
         PromptTagLengthEstimators["embedding"] = PromptTagLengthEstimators["preset"];
         PromptTagLengthEstimators["lora"] = PromptTagLengthEstimators["preset"];
+
+        PromptTagProcessors["setvar"] = (data, context) =>
+        {
+            string varname = context.PreData ?? "";
+            data = context.Parse(data);            
+            if (varname != "")
+            {
+                context.RandomVariables[varname] = data ?? "";
+            }
+            return data;
+        };
+        PromptTagLengthEstimators["setvar"] = (data) =>
+        {
+            return ProcessPromptLikeForLength(data);  //Should really be 0, since the setvar doesn't add anything to the prompt, but it accounts for one use of the variable

```

Comment:
setvar should not be 0, it does add things to the prompt.

Reply:
(if you intended for it to not add to the prompt, you implemented it wrong lol)

Reply:
I mean, *ideally* "setvar" would be 0 and we'd be getting the estimate on "var" instead, since "setvar" just passes through its contents after shoving a copy into a variable.

Reply:
setvar passes through its contents, therefore the length is the length of its contents